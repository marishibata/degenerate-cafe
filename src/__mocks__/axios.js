// 'use strict';
// module.exports = {
//   getNews: () => {
//     return Promise.resolve({
//       data: [
//         {
//           id: "0abcd",
//           title: "Lorem ipsum",
//           url: "https://messari.io/article/lorem-ipsum"
//         },
//         {
//           id: "1abcd",
//           title: "Testing",
//           url: "https://messari.io/article/testing"
//         }
//       ]
//     })
//   }
// }