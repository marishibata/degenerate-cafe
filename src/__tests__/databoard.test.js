import { jest } from '@jest/globals'
import React from 'react';
import Databoard from '../Components/databoard';
import { findByText, render, screen } from '@testing-library/react';
import axios from 'axios';

jest.mock('axios', () => {
  return {
    default: () => {
      return new Promise((resolve, reject) => {
        resolve(
          {
            data: {
              coins: [
                {
                  item: {
                    id: 0,
                    name: 'YIELD App'
                  }
                },
                {
                  item: {
                    id: 1,
                    name: 'SafeMoon'
                  }
                },
              ]
            }
          }
        )
      })
    }
  }
});

describe('fear and greed section', () => {
  it('Successfully renders the component', async () => {
    const { findByText } = render(<Databoard />);
    const result = await findByText('Trending On Coingecko');
    expect(result).toBeTruthy();
  });

  it('Successfully fetches coin name', async () => {
    const { findByText } = render(<Databoard />);
    const result = await findByText('YIELD App')
    expect(result).toBeTruthy();
  });

  it('Renders consistently', async () => {
    const { asFragment } = render(<Databoard />);
    expect(asFragment()).toMatchSnapshot();
  })
});
