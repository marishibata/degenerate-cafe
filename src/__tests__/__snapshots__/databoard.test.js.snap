// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fear and greed section Renders consistently 1`] = `
<DocumentFragment>
  <div
    class="datab-parent"
  >
    <div
      class="title"
    >
      <span>
        Trending On Coingecko
      </span>
    </div>
    <div
      class="par"
    />
    ;
  </div>
</DocumentFragment>
`;
